src/status_im/wallet/choose_recipient/core.cljs

```diff
@@ -121,15 +135,52 @@
       (assoc :ui/show-error (i18n/label :t/wallet-invalid-chain-id
                                         {:data data :chain current-chain-id})))))
 
+(fx/defn qr-scanner-cancel
+  {:events [:wallet.send/qr-scanner-cancel]}
+  [{db :db} _]
+  {:db (assoc-in db [:wallet/prepare-transaction :modal-opened?] false)})
+
+(fx/defn resolve-ens-addresses
+  {:events [:wallet.send/resolve-ens-addresses :wallet.send/qr-code-request-scanned]}
+  [{{:networks/keys [current-network] :wallet/keys [all-tokens] :as db} :db :as cofx}
+   uri origin]
+  (if-let [message (eip681/parse-uri uri)]
+  ;; first we get a vector of ens-names to resolve and a vector of paths of
+  ;; these names
+    (let [{:keys [paths ens-names]}
+          (reduce (fn [acc path]
+                    (let [address (get-in message path)]
+                      (if (ens/is-valid-eth-name? address)
+                        (-> acc
+                            (update :paths conj path)
+                            (update :ens-names conj address))
+                        acc)))
+                  {:paths [] :ens-names []}
+                  [[:address] [:function-arguments :address]])]
+      (if (empty? ens-names)
+      ;; if there is no ens-names, we dispatch request-uri-parsed immediately
+        (request-uri-parsed cofx message)
+        {::resolve-addresses
+         {:registry (get ens/ens-registries (ethereum/chain-keyword db))
+          :ens-names ens-names
+          :callback
+          (fn [addresses]
+            (re-frame/dispatch
+             [:wallet/request-uri-parsed
+            ;; we replace the ens-names at their path in the message by their
+            ;; actual address
+              (reduce (fn [message [path address]]
+                        (assoc-in message path address))
+                      message
+                      (map vector paths addresses))]))}}))
+    (cond-> {:db db}
+      true (assoc :ui/show-error (i18n/label :t/wallet-invalid-address {:data uri}))
+      (= origin :qr) (assoc :dispatch [:navigate-back]))))
```

Comment:
we don't have this in develop anymore, we don't pass origin in develop, so you can remove origin and this line

Reply:
Is this because the QR scanner is a modal now so we don't want to dispatch `navigate-back`?  I had it hardcoded as an origin of `:qr` in the `qr-scanner-result` method.

Reply:
yes, we close qr code screen in qr code handlers

Reply:
Got it.  Fixed in the latest commit by removing the `origin` parameter and updating the error messaging per your suggestion.

Reply:
have you pushed changes?

Reply:
:grimacing:  Sorry, committed locally and forgot to push to github.  The commits are out there now.